src/finch/finch_logic/interpreter.py

```diff
@@ -72,8 +72,11 @@ def __call__(self, node):
             for arg in args:
                 for idx, dim in zip(arg.idxs, arg.tns.shape, strict=True):
                     if idx in dims:
-                        if dims[idx] != dim:
-                            raise ValueError("Dimensions mismatched in map")
+                        if dims[idx] != dim and dim != 1:
+                            if dims[idx] == 1:  # broadcasting of 1-length dims
```

Comment:
I believe this behavior is supposed to be avoided, it breaks the "explicit broadcast" semantics of FinchLogic.